### PR TITLE
Fix analyze issues in gRPC ObjC lib

### PR DESCRIPTION
**Pull request by muxi**
Fixes analyze issues in Objective-C layer.

Partial fix of #16451.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,961,111      Total (>)      1,961,099

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,883,610      Total (<)     10,883,615

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,961,111      Total (>)      1,961,099

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,883,610      Total (>)     10,883,609

 No significant differences in binary sizes


```

File: src/objective-c/GRPCClient/private/GRPCHost.m [hunk 1]
```diff
@@ -137,7 +137,10 @@ - (nullable grpc_call *)unmanagedCallWithPath:(NSString *)path
                         completionQueue:queue];
 }
 
-- (NSData *)nullTerminatedDataWithString:(NSString *)string {
+- (NSData *)nullTerminatedDataWithString:(NSString *_Nullable)string {
```
Comment (dapengzhang0) [inline src/objective-c/GRPCClient/private/GRPCHost.m]: Also annotate return type with `_Nullable`?

Comment (dapengzhang0) [reply src/objective-c/GRPCClient/private/GRPCHost.m]: Per offline discussion, assert argument nonnull at usages of the method is better.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,965,565      Total (>)      1,965,327

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,888,291      Total (>)     10,888,230

 No significant differences in binary sizes


```

**Pull request merged**